Mathlib/Topology/Compactness/CompactlyGeneratedSpace.lean

```diff
@@ -0,0 +1,177 @@
+/-
+Copyright (c) 2024 Etienne Marion. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Etienne Marion
+-/
+import Mathlib.Topology.Category.CompactlyGenerated
+import Mathlib.Topology.Compactification.OnePoint
+
+/-!
+# Compactly generated topological spaces
+
+This file defines compactly generated topological spaces as a special case of
+`UCompactlyGeneratedSpaces`. A compactly generated space is a space such that a set `s` is closed
+if and only if for all compact Hausdorff space `K` and `f : K → X` continuous, `f ⁻¹' s` is closed
+in `K`. The typeclass `CompactlyGeneratedSpace` is defined by taking `K : Type u` when `X : Type u`,
+contrary to `UCompactlyGeneratedSpace` where spaces `K` can be taken in another universe.
+
+We prov basic properties and instances, and prove that a `SequentialSpace` is compactly generated,
+as well as Hausdorff `WeaklyLocallyCompactSpace`.
+
+## Main definitions
+
+* `CompactlyGeneratedSpace X`: the topology of `X` is coinduced by continuous maps coming from
+  compact Hausdorff spaces.
+
+## References
+
+* <https://en.wikipedia.org/wiki/Compactly_generated_space>
+* <https://ncatlab.org/nlab/files/StricklandCGHWSpaces.pdf>
+
+## Tags
+
+compactly generated space
+-/
+
+universe u v
+
+open TopologicalSpace Filter Topology Set
+
+variable {X : Type u} {Y : Type v} [TopologicalSpace X] [TopologicalSpace Y]
+
+/-- A topological space `X : Type u` is compactly generated if its topology is coinduced by
+continuous maps `f : K → X`, where `K : Type u` is compact Hausdorff. -/
+abbrev CompactlyGeneratedSpace (X : Type u) [TopologicalSpace X] : Prop :=
+  UCompactlyGeneratedSpace.{u} X
```

Comment:
The definition of `UCompactlyGeneratedSpace` could be moved to this file, along with the part of the API surrounding it that is not specific to the category `CompactlyGenerated`. Then the file `Topology.Category.CompactlyGenerated` should import this file.